lib/trino-orc/src/main/java/io/trino/orc/reader/SliceDictionaryColumnReader.java

```diff
@@ -365,4 +367,10 @@ public long getRetainedSizeInBytes()
     {
         return INSTANCE_SIZE;
     }
+
+    private void updateMemoryUsage()
+    {
+        systemMemoryContext.setBytes(sizeOf(nonNullValueTemp) + sizeOf(nonNullPositionList) + sizeOf(dictionaryData)
```

Comment:
`setBytes` is potentially expensive. How often is `setBytes` bytes called now?

Reply:
It dependens on the rate of read orc data, without memory count, the worker(especially small worker) maybe OOM, I think it is necessary and acceptable. 
Do you have any better idea?